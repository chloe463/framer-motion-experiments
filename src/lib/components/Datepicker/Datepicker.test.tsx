import { fireEvent, render } from "@testing-library/react";
import { Datepicker } from "./index";

describe("Dropdown", () => {
  it ("should render without crash", () => {
    const onChange = jest.fn();
    const { container } = render(<Datepicker value={new Date(2021, 0, 1)} onChange={onChange} />);
    const input = container.querySelector("div");
    expect(input).toBeTruthy();
  });

  it ("shoud opens options list on click", () => {
    const onChange = jest.fn();
    const { container } = render(
      <>
        <Datepicker
          value={new Date(2021, 0, 1)}
          onChange={onChange}
          calendarEntryPoint="calendar-entry-point"
        />
        <div id="calendar-entry-point"/>
      </>
    );

    const calendarBase = container.querySelector("div");
    expect(calendarBase).toBeTruthy();
    if (!calendarBase) throw new Error("No calendar base found");

    fireEvent.click(calendarBase);
    expect(container).toMatchSnapshot();

    const portalBase = container.querySelector("#calendar-entry-point");
    expect(portalBase).toBeTruthy();

    if (!portalBase) throw new Error("No portal base found");
    const dateButtons = portalBase.querySelectorAll("button");
    expect(dateButtons).toBeTruthy();
    if (!dateButtons) throw new Error("No date buttons found");
    expect(dateButtons.length).toBeGreaterThan(28);
  });
});
