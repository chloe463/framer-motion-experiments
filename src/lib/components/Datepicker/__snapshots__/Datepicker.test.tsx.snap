// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown shoud opens options list on click 1`] = `
<div>
  <div
    class="sc-bkzZxe iIWQPe"
    tabindex="0"
  >
    <label
      class="sc-dIUggk ODXsY datepicker-label"
    />
    <input
      class="sc-idOhPF sRzPA"
      type="hidden"
      value="2021-01-01"
    />
    <span
      class="sc-hHftDr heowOo"
    >
      2021/01/01
    </span>
    <span
      class="sc-dmlrTW imHFoD datepicker-bottom-border"
    />
  </div>
  <div
    id="calendar-entry-point"
  >
    <div
      class="sc-bdfBwQ iVirlS"
    >
      <span
        hidden=""
      />
      <div
        style="opacity: 0; transform: translateY(-16px) translateZ(0);"
      >
        <div
          aria-labelledby="date-picker"
          class="sc-gsTCUz hRzIb"
          role="dialog"
          style="position: fixed; transform: translate(0px, 0px);"
          tabindex="-1"
        >
          <header
            class="sc-dlfnbm kVrWBQ"
          >
            <div
              class="sc-hKgILt chiWrn"
            >
              <h2
                class="sc-jSgupP lkerYl"
                id="date-picker"
              >
                January
                 
                2021
              </h2>
              <button
                aria-label="Switch to year picker"
                class="sc-gKsewC dbAoBs"
                type="button"
              >
                <span
                  class="sc-iBPRYJ kfpHFw"
                />
              </button>
            </div>
            <div
              class="sc-eCssSg lffWUo"
            >
              <button
                aria-label="Show previous month"
                class="sc-gKsewC dbAoBs"
                type="button"
              >
                <span
                  class="sc-fubCfw jSOIpw"
                />
              </button>
              <button
                aria-label="Show next month"
                class="sc-gKsewC dbAoBs"
                type="button"
              >
                <span
                  class="sc-pFZIQ eepctN"
                />
              </button>
            </div>
          </header>
          <div
            class="sc-jrAGrp"
          >
            <div
              class="sc-kEjbxe grDMmH"
            >
              <span>
                S
              </span>
              <span>
                M
              </span>
              <span>
                T
              </span>
              <span>
                W
              </span>
              <span>
                T
              </span>
              <span>
                F
              </span>
              <span>
                S
              </span>
            </div>
            <div
              class="sc-iqHYGH ccGfrD"
            >
              <span />
              <span />
              <span />
              <span />
              <span />
              <button
                aria-label="2021-01-01"
                class="sc-crrsfI dWwSDJ"
                type="button"
              >
                1
              </button>
              <button
                aria-label="2021-01-02"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                2
              </button>
              <button
                aria-label="2021-01-03"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                3
              </button>
              <button
                aria-label="2021-01-04"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                4
              </button>
              <button
                aria-label="2021-01-05"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                5
              </button>
              <button
                aria-label="2021-01-06"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                6
              </button>
              <button
                aria-label="2021-01-07"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                7
              </button>
              <button
                aria-label="2021-01-08"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                8
              </button>
              <button
                aria-label="2021-01-09"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                9
              </button>
              <button
                aria-label="2021-01-10"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                10
              </button>
              <button
                aria-label="2021-01-11"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                11
              </button>
              <button
                aria-label="2021-01-12"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                12
              </button>
              <button
                aria-label="2021-01-13"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                13
              </button>
              <button
                aria-label="2021-01-14"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                14
              </button>
              <button
                aria-label="2021-01-15"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                15
              </button>
              <button
                aria-label="2021-01-16"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                16
              </button>
              <button
                aria-label="2021-01-17"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                17
              </button>
              <button
                aria-label="2021-01-18"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                18
              </button>
              <button
                aria-label="2021-01-19"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                19
              </button>
              <button
                aria-label="2021-01-20"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                20
              </button>
              <button
                aria-label="2021-01-21"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                21
              </button>
              <button
                aria-label="2021-01-22"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                22
              </button>
              <button
                aria-label="2021-01-23"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                23
              </button>
              <button
                aria-label="2021-01-24"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                24
              </button>
              <button
                aria-label="2021-01-25"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                25
              </button>
              <button
                aria-label="2021-01-26"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                26
              </button>
              <button
                aria-label="2021-01-27"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                27
              </button>
              <button
                aria-label="2021-01-28"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                28
              </button>
              <button
                aria-label="2021-01-29"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                29
              </button>
              <button
                aria-label="2021-01-30"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                30
              </button>
              <button
                aria-label="2021-01-31"
                class="sc-crrsfI gdRmrE"
                type="button"
              >
                31
              </button>
              <button
                class="sc-crrsfI gdRmrE"
                disabled=""
                type="button"
              >
                1
              </button>
              <button
                class="sc-crrsfI gdRmrE"
                disabled=""
                type="button"
              >
                2
              </button>
              <button
                class="sc-crrsfI gdRmrE"
                disabled=""
                type="button"
              >
                3
              </button>
              <button
                class="sc-crrsfI gdRmrE"
                disabled=""
                type="button"
              >
                4
              </button>
              <button
                class="sc-crrsfI gdRmrE"
                disabled=""
                type="button"
              >
                5
              </button>
              <button
                class="sc-crrsfI gdRmrE"
                disabled=""
                type="button"
              >
                6
              </button>
            </div>
          </div>
          <footer
            class="sc-kstrdz dOREyJ"
          >
            <button
              class="sc-hBEYos sc-fodVxV bAQEfG"
              type="button"
            >
              Cancel
            </button>
            <button
              class="sc-hBEYos sc-fFubgz bAQEfG"
              type="button"
            >
              OK
            </button>
          </footer>
        </div>
      </div>
      <span
        hidden=""
      />
    </div>
  </div>
</div>
`;
